### ISSUE TITLE
Restart menu item missing in 5.1.6

### ISSUE THREAD

**Issue by Ketrel** [MrAlex94/Waterfox#3039]
### What happened?

As of 5.1.6, the "Restart (Developer)" item in the file menu item is missing.

This was not tested on Firefox, as I don't recall this being a Firefox feature to begin with

### Reproducible?

- [x] I have tested this on Firefox.

### Version

G5

### What platform are you seeing the problem on?

Windows

EDIT: it looks like it's incomplete unless I mark that I tested in firefox, which doesn't make any sense in this context, but firefox doesn't have it either, but it didn't used to like waterfox did.

**Comment by Ketrel** [comment]
This continues to be the case in 5.1.7

**Comment by thunderstone135** [comment]
I can confirm this. Even the keyboard shortcut Ctrl + Alt + R is gone.

**Comment by Hurri04** [comment]
In the Settings menu there's an option "Show restart button in PanelUI" (or something similar, translated), indicating that the underlying method for this is still available.

However, since I've hidden the hamburger menu in the top right via some css, this change is indeed semi-optimal. 
Being able to quickly restart when experimenting with different addons by pressing `ALT`, then selecting `File` → `Restart` was pretty neat.

Any word on why this option was removed and if we can get it back?

P.S.: I tried looking for installers for older versions [here](https://cdn.waterfox.net/releases/) (as indicated on the [official download page](https://www.waterfox.net/download/)) but it seems like this has been neglected and there is not a single G5 version on there, leaving (potentially sketchy) third-party websites as the only option?

**Comment by MrAlex94** [comment]
On Windows:

![image](https://github.com/WaterfoxCo/Waterfox/assets/4728920/1f1770c1-239a-4b95-81f3-453004a29fd8)

On macOS:

![image](https://github.com/WaterfoxCo/Waterfox/assets/4728920/2f6620e5-6cfa-49e3-a609-99f9c108653c)

The keyboard shortcut will be missing, as `MOZILLA_OFFICIAL` was defined as it should have been in the first release of G5. Essentially, it disables all "developer" focused options, and the keyboard shortcut was part of that. 

Can I just confirm it's just the keyboard shortcut missing?

**Comment by Hurri04** [comment]
Does this `MOZILLA_OFFICIAL`flag affect Windows and macOS differently? Because on Windows the "Restart" menu item is missing from the `File` menu, as opposed to your macOS screenshot: 
![RestartMissing](https://github.com/WaterfoxCo/Waterfox/assets/34166588/707bd6af-64e3-47f4-b661-511e9fee4fcf)

The "Restart" option from your Windows screenshot seems to be controlled by the first checkbox here: 
![RestartSetting](https://github.com/WaterfoxCo/Waterfox/assets/34166588/d7dd6939-e495-45a0-943b-2e04afbfca75)

**Comment by thunderstone135** [comment]
> Can I just confirm it's just the keyboard shortcut missing?

Both keyboard shortcut (Ctrl + Alt + R) and File menu item [File > Restart (Developer)] are gone.

**Comment by MrAlex94** [comment]
> > Can I just confirm it's just the keyboard shortcut missing?
> 
> Both keyboard shortcut (Ctrl + Alt + R) and File menu item [File > Restart (Developer)] are gone.

Yes, Restart (Developer) was not meant to be shown at all, it is meant for "local" builds.

@Hurri04 not sure, I suppose the File menu is kind of forgotten on Windows, which is why we added it to the Panel menu instead.

**Comment by thunderstone135** [comment]
> Yes, Restart (Developer) was not meant to be shown at all, it is meant for "local" builds.

If the file menu item will no longer be available in Waterfox, can you at least restore the keyboard shortcut?

**Comment by thunderstone135** [comment]
**Update:**

I noticed that this entry is missing in omni.ja 👇 

```/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

/**
 * Extra features for local development. This file isn't loaded in
 * non-local builds.
 */

var DevelopmentHelpers = {
  init() {
    this.quickRestart = this.quickRestart.bind(this);
    this.addRestartShortcut();
  },

  quickRestart() {
    Services.obs.notifyObservers(null, "startupcache-invalidate");

    let env = Cc["@mozilla.org/process/environment;1"].getService(
      Ci.nsIEnvironment
    );
    env.set("MOZ_DISABLE_SAFE_MODE_KEY", "1");

    Services.startup.quit(
      Ci.nsIAppStartup.eAttemptQuit | Ci.nsIAppStartup.eRestart
    );
  },

  addRestartShortcut() {
    let command = document.createXULElement("command");
    command.setAttribute("id", "cmd_quickRestart");
    command.addEventListener("command", this.quickRestart, true);
    document.getElementById("mainCommandSet").prepend(command);

    let key = document.createXULElement("key");
    key.setAttribute("id", "key_quickRestart");
    key.setAttribute("key", "r");
    key.setAttribute("modifiers", "accel,alt");
    key.setAttribute("command", "cmd_quickRestart");
    document.getElementById("mainKeyset").prepend(key);

    let menuitem = document.createXULElement("menuitem");
    menuitem.setAttribute("id", "menu_FileRestartItem");
    menuitem.setAttribute("key", "key_quickRestart");
    menuitem.setAttribute("label", "Restart (Developer)");
    menuitem.addEventListener("command", this.quickRestart, true);
    document.getElementById("menu_FilePopup").appendChild(menuitem);
  },
};
```

This file is `browser-development-helpers.js` which was located in `omni.ja\chrome\browser\content\browser`.
I just really want the Ctrl + Alt + R keyboard shortcut back in Waterfox.

**Comment by keithbowes** [comment]
> **Update:**
> 
> I noticed that this entry is missing in omni.ja 👇
> 
> ```
>  * License, v. 2.0. If a copy of the MPL was not distributed with this
>  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
> 
> /**
>  * Extra features for local development. This file isn't loaded in
>  * non-local builds.
>  */
> 
> var DevelopmentHelpers = {
>   init() {
>     this.quickRestart = this.quickRestart.bind(this);
>     this.addRestartShortcut();
>   },
> 
>   quickRestart() {
>     Services.obs.notifyObservers(null, "startupcache-invalidate");
> 
>     let env = Cc["@mozilla.org/process/environment;1"].getService(
>       Ci.nsIEnvironment
>     );
>     env.set("MOZ_DISABLE_SAFE_MODE_KEY", "1");
> 
>     Services.startup.quit(
>       Ci.nsIAppStartup.eAttemptQuit | Ci.nsIAppStartup.eRestart
>     );
>   },
> 
>   addRestartShortcut() {
>     let command = document.createXULElement("command");
>     command.setAttribute("id", "cmd_quickRestart");
>     command.addEventListener("command", this.quickRestart, true);
>     document.getElementById("mainCommandSet").prepend(command);
> 
>     let key = document.createXULElement("key");
>     key.setAttribute("id", "key_quickRestart");
>     key.setAttribute("key", "r");
>     key.setAttribute("modifiers", "accel,alt");
>     key.setAttribute("command", "cmd_quickRestart");
>     document.getElementById("mainKeyset").prepend(key);
> 
>     let menuitem = document.createXULElement("menuitem");
>     menuitem.setAttribute("id", "menu_FileRestartItem");
>     menuitem.setAttribute("key", "key_quickRestart");
>     menuitem.setAttribute("label", "Restart (Developer)");
>     menuitem.addEventListener("command", this.quickRestart, true);
>     document.getElementById("menu_FilePopup").appendChild(menuitem);
>   },
> };
> ```
> 
> This file is `browser-development-helpers.js` which was located in `omni.ja\chrome\browser\content\browser`. I just really want the Ctrl + Alt + R keyboard shortcut back in Waterfox.

Thanks. I was searching for a way to remove the annoying 'Restart (developer)' menu item from LibreWolf and your comment helped me. I just added the following to my userChrome.css:
```
#menu_FileRestartItem {
  display: none;
}
```

It does nothing about [the keyboard shortcut conflicting with that  for "Enter Reader View"](https://bugzilla.mozilla.org/show_bug.cgi?id=1829544), but small problems (especially since I never use reader view).
